rclcpp_lifecycle/test/benchmark/benchmark_lifecycle_client.cpp

```diff
@@ -203,7 +203,7 @@ class BenchmarkLifecycleClient : public performance_test_fixture::PerformanceTes
     executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
     executor->add_node(lifecycle_node->get_node_base_interface());
     executor->add_node(lifecycle_client->get_node_base_interface());
-    spinner_ = std::thread(&rclcpp::executors::SingleThreadedExecutor::spin, executor);
+    spinner_ = std::thread([this]() {executor->spin();});
```

Comment:
Why is this change needed?

Reply:
I see it doesn't compile... this is the kind of change that suggests we should rethink the approach. The old style of calling it should still work I think. I'll leave it, but it's another point against this approach in my mind.